hclasses/CReporter.cpp

```diff
@@ -0,0 +1,73 @@
+#include "CReporter.h"
+#include "CDebugger_H2.h"
+
+#include <fstream>
+#include <sstream>
+#include <string>
+
+using Severities = CReportMessage::Severities;
+using Categories = CReportMessage::Categories;
+
+CReporter *CReporter::instance = NULL;
+
+CReporter::CReporter(bool report_and_log)
+    : messageList(), writelog(report_and_log), _filename() {
+    // Empty at this point
+}
+
+CReporter *CReporter::getInstance() {
+  if (instance == NULL) {
+      if (REPORT_AND_LOG == true)
+          instance = new CReporter(REPORT_AND_LOG);
+      else
+          instance = new CReporter();
+  }
+
+  return instance;
+}
+
+void CReporter::addMessage(const CT::string message, CReportMessage::Severities severity,
+                           CReportMessage::Categories category, CT::string documentationLink, const char *file,
+                           int line, const char *className) {
+
+  if(this->writelog) {
+    writeMessageToLog(message, severity, file, line, className);
```

Comment:
Are errors really shown in default ADAGUC logs (without reporting enabled), also see removal of CImageDataWriter.cpp, 3361 (CDBError)

Reply:
Yes. Wrote a test to verify. Commit a484eb4677fe318b57decea33b506f547f31cebe